src/view/com/post-thread/PostThread.tsx

```diff
@@ -215,24 +254,52 @@ function PostThreadLoaded({
           offset: pageY,
         })
       }
-      if (isNative) {
-        highlightedPostRef.current?.measure(
-          (_x, _y, _width, _height, _pageX, pageY) => {
-            onMeasure(pageY)
-          },
-        )
-      } else {
-        // Measure synchronously to avoid a layout jump.
-        const domNode = highlightedPostRef.current
-        if (domNode) {
-          const pageY = (domNode as any as Element).getBoundingClientRect().top
-          onMeasure(pageY)
-        }
+      // Measure synchronously to avoid a layout jump.
+      const domNode = highlightedPostRef.current
+      if (domNode) {
+        // @ts-ignore web only
+        const pageY = (domNode as any as Element).getBoundingClientRect().top
+        onMeasure(pageY)
       }
       needsScrollAdjustment.current = false
     }
   }, [thread])
 
+  // HACK
+  // Until we can get maintainVisibleContentPosition to work regardless of how many posts we have rendered at the
+  // top, this will simulate a "load" when we reach the top, to allow for the next items to render without jumping
+  const onStartReached = React.useCallback(() => {
+    // Get the first post. We need to get the second item in the array if the first one is the TOP_COMPONENT
+    let first = posts?.[0]
+    if (first === TOP_COMPONENT) {
+      first = posts?.[1]
+    }
+
+    // We do nothing in these situations
+    // - We are already prepending
+    // - This is not a thread post
+    // - There are no more parents to show
+    // - The top post is the highlighted post
+    if (
+      isPrepending.current ||
+      !isThreadPost(first) ||
+      !first.parent ||
+      first.ctx.isHighlightedPost
+    ) {
+      return
+    }
```

Comment:
1. We don't want to update `topPageCount` multiple times (`onStartReached` doesn't get debounced, so it's possible it will fire multiple times once the user reaches the top). So, we use a ref to keep track of whether we're awaiting new items to render. Once they do, we will set the ref back to false.
2. `isThreadPost` just gives us the type we need.
3. `!first.parent` checks if there is anything else for us to add. If there isn't then we don't need to do anything and can just return.
4. Similar to `!first.parent`, I think we can remove this now that I think about it.